### PR TITLE
Fixes #2878 : Using multi-level comments display.

### LINKED ISSUE #2425: Comments sheet sometimes exceeds the maximum height
### Steps to reproduce

1. Open the following video in LibreTube: https://piped.video/watch?v=3PcUzzpkHMk
2. Open the comments sheet
3. Click on the first comment with 21 replies 

### Expected behavior

The comments sheet should unfold the replies and stay at the same size.

### Actual behavior

The comments sheet somehow gets fullscreen.
I have no clue what could be causing this. Probably the huge amount of replies and the fact that it is the first comment??

### LibreTube version

0.9.1

### Android version

Android 13 QPR1

### Other details

I use GrapheneOS.
If you have issues reproducing this, I am open to sending a video or logs.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #2876: Do not darken video while reading comments
### Describe your suggested feature

At the moment, opening the "comments" view below a video darkens everything outside of it. While I see the point of that design, it makes it unnecessarily difficult to read while watching at the same time. The proprietary YouTube app deosn't do this either. It would be very nice if this was at least an option able to be turned off.

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #2878: Using multi-level comments display.
### Describe your suggested feature

Currently the LibreTube displays the comments and replies in the same level as Instagram. It would be better if we could make it multi-level as YouTube by showing the replies on a comment click, in a nested fragment. When a comment has a lot of replies then it becomes confusing between comments and replies. 

We should also get inspiration from YouTube for colors and font sizes to make the usernames, times and comments distinct.

### Other details

Screenshot_2023-01-27-15-42-09-71_7af1185115278aeb9699bfd161691b7e

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [x] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by faisalcodes**
- Comments and replies are now displayed in different fragments making the experience more friendly. (Issue #2878)
- Video is not darkened when comments dialog is opened. (closes #2876)
- closes #2425

- Although video is not dimmed when comments dialog is opened, clicks on the video has no effect until dialog is closed. I could not fix this.

### CODE REVIEW

File: app/src/main/java/com/github/libretube/api/obj/Comment.kt [hunk 1]
```diff
@@ -6,7 +6,7 @@ import kotlinx.serialization.Serializable
 data class Comment(
     val author: String,
     val commentId: String,
-    val commentText: String,
+    val commentText: String?,
```
Comment (faisalcodes) [inline app/src/main/java/com/github/libretube/api/obj/Comment.kt:9]: Server was returning null for some comments and kotlinx-serializer could not succeed.

**Comment by Bnyro** [comment]
Awesome work!

**Review by Bnyro — APPROVED** [review]

**Comment by ARBoyGo** [comment]
السَّلاَمُ عَلَيْكُمْ وَرَحْمَةُ اللهِ وَبَرَكَاتُهُ 
Assalamu Alaikum Wa Rahmatullahi Wabarakatuh | Peace be upon you and God's mercy and blessings be upon you. 

Thanks Faisal Brother 💕 
This looks amazing. 
I feel like YouTube haha. 
Hoping for more PR from you. I saw your first PR was seek bar, I was saying to bnyro you are seek expert, But you are good at all actually 💕. 

Where can I connect you? Matrix? Irc? Discord? Telegram?

**Comment by ARBoyGo** [comment]
السَّلاَمُ عَلَيْكُمْ وَرَحْمَةُ اللهِ وَبَرَكَاتُهُ 
Assalamu Alaikum Wa Rahmatullahi Wabarakatuh | Peace be upon you and God's mercy and blessings be upon you.

Thanks @faisalcodes Bhai/Brother 💕
This looks amazing.
I feel like YouTube haha.
Hoping for more PR from you. I saw your first PR was seek bar, I was saying to bnyro you are seek expert, But you are good at all actually 💕.

Where can I connect you? Matrix? Irc? Discord? Telegram?

**Pull request merged**